File: src/test/java/com/animalfarm/admin/repository/TradeHistoryRepositoryTests.java
package com.animalfarm.admin.repository;

import com.animalfarm.admin.collection.TradeHistory;
import lombok.extern.slf4j.Slf4j;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertTrue;

@SpringBootTest
@Slf4j
public class TradeHistoryRepositoryTests {
    @Autowired
    private TradeHistoryRepository tradeHistoryRepository;

    @Test
    public void findAllTest() {
        Page<TradeHistory> page = tradeHistoryRepository.findAll(Pageable.ofSize(10));
        assertThat(page.getTotalElements()).isEqualTo(5);
        page.get().map(TradeHistory::toString).forEach(log::info);
    }

    @Test
    public void findAllByKeywordTest() {
        Page<TradeHistory> page = tradeHistoryRepository.findAllWithKeyword(Pageable.ofSize(10), null);
        assertThat(page.getTotalElements()).isEqualTo(5);
        page.get().map(TradeHistory::toString).forEach(log::info);
    }

    @Test
    public void regexTest() {
        assertTrue("keyword".matches(".*keyword.*"));
        assertTrue("123keyword123".matches(".*keyword.*"));
        assertTrue("123  keyword   123".matches(".*keyword.*"));
        assertTrue("   keyword   123".matches(".*keyword.*"));
    }
}

File: src/test/kotlin/com/animalfarm/admin/repository/TradeHistoryRepositoryTests.kt
package com.animalfarm.admin.repository

import com.animalfarm.admin.collection.TradeHistory
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.annotation.Rollback

@SpringBootTest
@Rollback(false)
class TradeHistoryRepositoryTests() {
    @Autowired
    lateinit var log: Logger

    @Autowired
    lateinit var tradeHistoryRepository: TradeHistoryRepository

    @Test
    fun findTest() {
        val uuid = "005cd92d-cf8a-4267-bfc5-50d3f039d50f"
        log.info("=============================================")

        tradeHistoryRepository.findAllByAcceptor_UuidOrRequester_Uuid(uuid, uuid)
            .map(TradeHistory::toString)
            .forEach(log::info)

        tradeHistoryRepository.findAll().map(TradeHistory::toString).forEach(log::info)
    }
}